packages/core-data/src/entity-provider.js

```diff
@@ -153,16 +150,36 @@ export function useEntityProp( kind, name, prop, _id ) {
 export function useEntityBlockEditor( kind, name, { id: _id } = {} ) {
 	const providerId = useEntityId( kind, name );
 	const id = _id ?? providerId;
-	const { content, editedBlocks, meta } = useSelect(
+	const { blocks, meta } = useSelect(
 		( select ) => {
 			if ( ! id ) {
 				return {};
 			}
-			const { getEditedEntityRecord } = select( STORE_NAME );
+			const { getEditedEntityRecord, getEntityRecord } =
+				select( STORE_NAME );
 			const editedRecord = getEditedEntityRecord( kind, name, id );
+
+			let editedBlocks = editedRecord.blocks;
+
+			if ( ! editedBlocks ) {
+				if (
+					editedRecord.content &&
+					typeof editedRecord.content !== 'function'
+				) {
+					// Attach the cache to the original record.
+					const entityRecord = getEntityRecord( kind, name, id );
+					editedBlocks = parsedBlocksCache.get( entityRecord );
+					if ( ! editedBlocks ) {
+						editedBlocks = parse( editedRecord.content );
+						parsedBlocksCache.set( entityRecord, editedBlocks );
+					}
+				} else {
+					editedBlocks = EMPTY_ARRAY;
+				}
+			}
```

Comment:
Why did you decide to move this code inside useSelect calls. I'm worried about parse within useSelect, can we move it to the previous useMemo instead?

Reply:
Sure, I guess I can pass the entity record to the component

Reply:
Fixed